crates/matrix-sdk-crypto/src/machine/mod.rs

```diff
@@ -1465,11 +1462,32 @@ impl OlmMachine {
                 // No - use the existing data.
                 session.sender_data.clone()
             }
+        } else {
+            session.sender_data.clone()
         };
 
         Ok(sender_data_to_verification_state(sender_data, session.has_been_imported()))
     }
 
+    /// Whether we should recalculate the Megolm sender's data, given the
+    /// current sender data. We only want to recalculate if it might increase
```

Comment:
I feel like this is missing context. It surely can't be the case that we *always* want to recalculate senderdata -- there must be some event that prompts it? In other words, when is this method called?

Reply:
This is used when we decrypt a message, and its logic is tied to those circumstances. A free function or a function definition inside `get_or_update_verification_state` might be better?

Reply:
Moved to an inner function in 3a84196573